### PR TITLE
Add dyson finance adapter

### PR DESCRIPTION
**Pull request by ytwang00**

### CODE REVIEW

**Review by 0xroll — APPROVED** [review]

**Comment by 0xroll** [comment]
@ytwang00 hi can you fix to pass the test check?

also when users deposit , are they given a receipt token? if yes we should also be tracking these in adapter

**Comment by ytwang00** [comment]
@0xroll 
I fixed the code.
And, we don't use receipt tokens in this protocol.

File: adapters/dysonfinance/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+                        decimals: parseFloat(pairData.token0Decimals)
+				  }
+				: {
+						tokenAddress: pairData.token1Address,
+						tokenSymbol: pairData.token1Name,
+                        decimals: parseFloat(pairData.token1Decimals)
+				  };
+		const id = `${element.user}-${depositToken.tokenAddress}`
+        let position: SumPosition | undefined = userPositionMap.get(id)
+        const depositAmt = parseFloat(formatUnits(element.depositAmt, depositToken.decimals))
```
Comment (0xroll) [inline adapters/dysonfinance/src/index.ts:172]: are these currently formatting the numbers to real numbers?

we would want raw numbers here

Comment (0xroll) [reply adapters/dysonfinance/src/index.ts:172]: @ytwang00

Comment (ytwang00) [reply adapters/dysonfinance/src/index.ts:172]: Yes, it's formatted to real numbers.
I'll fix it very soon.

**Comment by ytwang00** [comment]
@0xroll 
Pagination issue Fixed

DYSON_POSITIONS_QUERY => fetching all data with pagination 
DYSON_POOLS_QUERY => We won't have more than 10 pools, no need to use pagination

**Review by 0xroll — APPROVED** [review]

File: adapters/dysonfinance/src/index.ts [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+            user
+            pair
+            depositType
+            depositAmt
+            token0Amt
+            token1Amt
+            isWithdrawed
+          }
+    }
+`;
```
[the remark this answers is not in the document]
Comment (ytwang00) [reply adapters/dysonfinance/src/index.ts]: May 5th and two days ago? 
I must have missed that...

**Pull request merged**